ci/docker/conda-cpp.dockerfile

```diff
@@ -22,13 +22,18 @@ FROM ${repo}:${arch}-conda
 COPY ci/scripts/install_minio.sh /arrow/ci/scripts
 RUN /arrow/ci/scripts/install_minio.sh latest /opt/conda
 
+# Unless overriden use Python 3.10
+# Google GCS fails building with Python 3.11 at the moment.
+ARG python=3.10
+
 # install the required conda packages into the test environment
 COPY ci/conda_env_cpp.txt \
      ci/conda_env_gandiva.txt \
      /arrow/ci/
 RUN mamba install -q -y \
         --file arrow/ci/conda_env_cpp.txt \
         --file arrow/ci/conda_env_gandiva.txt \
+        python=${python} \
         compilers \
         doxygen \
         valgrind && \
```

Comment:
Could you keep this list in alphabetical order?

Reply:
Done, thanks @kou